paddle/cinn/hlir/dialect/operator/transforms/group_merge/convert_static_dim_to_dynamic_pass.cc

```diff
@@ -0,0 +1,362 @@
+#include "paddle/cinn/hlir/dialect/operator/transforms/group_merge/convert_static_dim_to_dynamic_pass.h"
+#include "paddle/pir/dialect/shape/utils/dim_expr.h"
+#include "paddle/pir/dialect/shape/utils/shape_utils.h"
+#include "paddle/cinn/common/dim_expr_util.h"
+
+namespace cinn::dialect::ir {
+
+namespace {
+
+PD_DEFINE_string(cinn_convert_static_dim_to_dynamic);
+
+template <typename DoEachT>
+void ForEachRawStaticDimToDyanmicPair(const DoEachT& DoEach) {
+  const std::string& env_var = FLAGS_cinn_convert_static_dim_to_dynamic;
+  size_t start = 0;
+  while (start != std::string::npos) {
+    size_t end = env_var.find(",", start);
+    DoEach(env_var.substr(start, end));
+    start = end + (end != std::string::npos);
+  }
+}
+
+std::optional<std::pair<int64_t, std::string>> ParseRawStaticDimToDyanmicPair(const std::string& raw_pair) {
+  size_t pos = raw_pair.find(":", 0);
+  if (pos == std::string::npos) return std::nullopt;
+  long long constant = 0;
+  try {
+    constant = std::stoll(raw_pair.substr(0, pos), nullptr);
+  } catch (const std::invalid_argument&) {
+    return std::nullopt;
+  }
+  if (constant <= 0) return std::nullopt;
+  std::string symbol = raw_pair.substr(pos + 1, -1);
+  if (symbol == "") return std::nullopt;
+  const auto& IsWord = [&](const char ch) {
+    if (ch >= 'a' && ch <= 'z') return true;
+    if (ch >= 'A' && ch <= 'Z') return true;
+    if (ch == '_') return true;
+    return false;
+  };
+  const auto& IsDigit = [&](const char ch) {
+    return ch >= '0' & ch <= '9';
+  };
+  if (!IsWord(symbol[0])) return std::nullopt;
+  for (int i = 1; i < symbol.size(); ++i) {
+    if (!(IsWord(symbol[i]) || IsDigit(symbol[i]))) return std::nullopt;
+  }
+  return std::pair{int64_t{constant}, symbol};
+}
+
+std::unordered_map<int64_t, std::string> GetStaticDimToDyanmicFromFlag() {
+  std::unordered_map<int64_t, std::string> map{};
```

Comment:
Can we remove `{}`? Nothing, just pretty.

Reply:
done